noxfile.py

```diff
@@ -115,3 +124,71 @@ def docfx(session):
         os.path.join("docs", ""),
         os.path.join("docs", "_build", "html", ""),
     )
+
+
+@nox.session(python=DEFAULT_PYTHON_VERSION)
+def lint(session):
+    """Run linters.
+
+    Returns a failure if the linters find linting errors or
+    sufficiently serious code quality issues.
+    """
+    session.install(FLAKE8_VERSION, BLACK_VERSION)
+    session.run(
+        "black",
+        "--check",
+        *LINT_PATHS,
+    )
+    session.run("flake8", "google", "tests")
+
+
+@nox.session(python=DEFAULT_PYTHON_VERSION)
+def lint_setup_py(session):
+    """Verify that setup.py is valid (including an RST check)."""
+    session.install("docutils", "pygments")
+    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
+
+
+@nox.session(python=DEFAULT_PYTHON_VERSION)
+def blacken(session):
+    session.install(BLACK_VERSION)
+    session.run(
+        "black",
+        *LINT_PATHS,
+    )
+
+
+@nox.session(python=DEFAULT_PYTHON_VERSION)
+def format(session):
+    session.install(BLACK_VERSION, ISORT_VERSION)
+    # Sort imports in strict alphabetical order.
+    session.run(
+        "isort",
+        "--fss",
+        *LINT_PATHS,
+    )
+    session.run(
+        "black",
+        *LINT_PATHS,
+    )
+
+
+def unit(session):
+    install_unittest_dependencies(session)
+    session.run(
+        "py.test",
+        "--quiet",
+        os.path.join("tests", "unit"),
```

Comment:
We don't have a unit directory

Reply:
@averikitsch in the follow-up PR I added one, but didn't want to introduce it in this already huge PR.